File: helfomat-core/src/main/java/de/helfenkannjeder/helfomat/core/organization/event/OrganizationEditLogoEvent.java
package de.helfenkannjeder.helfomat.core.organization.event;

import de.helfenkannjeder.helfomat.core.organization.Organization;
import de.helfenkannjeder.helfomat.core.organization.OrganizationEventVisitor;
import de.helfenkannjeder.helfomat.core.organization.OrganizationId;
import de.helfenkannjeder.helfomat.core.picture.PictureId;

/**
 * @author Valentin Zickner
 */
@SuppressWarnings({"WeakerAccess", "CanBeFinal"})
public class OrganizationEditLogoEvent extends OrganizationEditEvent {
    private PictureId logo;

    protected OrganizationEditLogoEvent() {
    }

    public OrganizationEditLogoEvent(OrganizationId organizationId, PictureId logo) {
        super(organizationId);
        this.logo = logo;
    }

    public PictureId getLogo() {
        return logo;
    }

    @Override
    public Organization.Builder applyOnOrganizationBuilder(Organization.Builder organization) {
        return organization.setLogo(logo);
    }

    @Override
    public <T> T visit(OrganizationEventVisitor<T> visitor) {
        return visitor.visit(this);
    }
}

File: helfomat-core/src/main/kotlin/de/helfenkannjeder/helfomat/core/organization/event/OrganizationEditLogoEvent.kt
package de.helfenkannjeder.helfomat.core.organization.event

import de.helfenkannjeder.helfomat.core.organization.Organization
import de.helfenkannjeder.helfomat.core.organization.OrganizationEventVisitor
import de.helfenkannjeder.helfomat.core.organization.OrganizationId
import de.helfenkannjeder.helfomat.core.picture.PictureId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditLogoEvent(
    override val organizationId: OrganizationId,
    val logo: PictureId?
) : OrganizationEditEvent(organizationId) {

    override fun applyOnOrganizationBuilder(organizationBuilder: Organization.Builder?): Organization.Builder? {
        return organizationBuilder?.setLogo(logo)
    }

    override fun <T> visit(visitor: OrganizationEventVisitor<T>): T {
        return visitor.visit(this)
    }

}